acceptancetests/assess_cross_model_relations.py

```diff
@@ -104,6 +105,62 @@ def ensure_cmr_offer_management(client):
         assert_offer_can_be_deleted(management_model, offer_url)
 
 
+def ensure_cmr_offer_consumption_and_relation(first_client, second_client):
+    """Ensure offers can be consumed by another model.
+
+    :param first_client: ModelClient model that will be the source of the
+      offer.
+    :param second_client: ModelClient model that will consume the offered
+      application endpoint.
+    """
+    with temporary_model(first_client, 'relation-source') as source_client:
+        with temporary_model(second_client, 'relation-sink') as sink_client:
+            offer_url, offer_name = deploy_and_offer_db_app(source_client)
+            assert_relating_to_offer_succeeds(sink_client, offer_url)
+            assert_saas_url_is_correct(sink_client, offer_name, offer_url)
+
+
+def ensure_user_can_consume_offer(first_client, second_client):
+    """Ensure admin is able to grant a user access to an offer.
+
+    Almost the same as `ensure_cmr_offer_consumption_and_relation` except in
+    this a user is created on all controllers (might be a single controller
+    or 2) with the permissions of 'login' for the source controller and 'write'
+    for the model into which the user will deploy an application to consume the
```

Comment:
The user doesn't deploy an application to consume the offer.
They can: 
- consume the offer (this just adds a SAAS entry in status and reflects the state of the offer in the offering model)
- deploy a local application and relate it to the offer (or relate it to an offer that has already been consumed).

ie, either

$ juju deploy mediawiki
$ juju consume some-offer-url hosted-mysql
$ juju relate mediawiki:db hosted-mysql

or

$ juju deploy mediawiki
$ juju relate mediawiki:db some-offer-url

In either case, you get a consumed application / SAAS entry in status. And a relation to that SAAS entry.